### PR TITLE
Add basic SweetTableViewController

### PR DESCRIPTION
**Pull request by elland**
Same as SweetCollectionViewControler. (#8)

### CODE REVIEW

File: Demo/TableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+class TableCell: UITableViewCell {
```
Comment (3lvis) [inline Demo/TableViewController.swift]: We have TableCell and no TableController?

Comment (3lvis) [inline Demo/TableViewController.swift]: Should we rename `TableViewController` to `TableController` instead? Or rename to `TableViewCell`

Comment (elland) [reply Demo/TableViewController.swift]: You lost me there. What do you mean?

File: Demo/TableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+}
+
+class TableViewController: SweetTableViewController {
+    lazy var data: [String] = {
```
Comment (3lvis) [inline Demo/TableViewController.swift]: `let data = ["Some", "Stringing", "Going", "On"]`

I think you could use this instead.

File: Demo/TableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+}
+
+extension TableViewController: UITableViewDataSource {
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.data.count
+    }
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(TableCell.reuseIdentifier, forIndexPath: indexPath) as! TableCell
```
Comment (3lvis) [inline Demo/TableViewController.swift]: We could steal/borrow this from here: http://alisoftware.github.io/swift/generics/2016/01/06/generic-tableviewcells/

``` swift
func dequeueReusableCell<T: Reusable>(indexPath indexPath: NSIndexPath) -> T {
  return self.dequeueReusableCellWithIdentifier(T.reuseIdentifier, forIndexPath: indexPath) as! T
}
```

Comment (3lvis) [inline Demo/TableViewController.swift]:
``` swift
let cell = tableView.dequeueReusableCell(indexPath: indexPath) as MyCustomCell
```

Comment (elland) [reply Demo/TableViewController.swift]: I like the idea here. 👍

Comment (elland) [reply Demo/TableViewController.swift]: But this is really too much _magic_ for my taste.

`Now thanks to Swift’s type inference, this method will use the call-site context to infer the actual type of T, and thus this type can be somehow seen as… “retro-injected” into the implementation! ✨`

Comment (3lvis) [reply Demo/TableViewController.swift]: It's pretty normal Swift, generics it's a pretty common concept. Should probably be fine.

Comment (elland) [reply Demo/TableViewController.swift]: The Generics syntax is just a clusterfuck to me. Also there it's not really clear to me how the type-inference system _retro-injects_ the class into the implementation. Maybe it's just me. 💭 

Thinking some more.

Comment (3lvis) [reply Demo/TableViewController.swift]: That's just fancy talk for avoiding the forced unwrapping

Comment (elland) [reply Demo/TableViewController.swift]: No, I mean it knows the class type because of the `<T: Reusable>` added to the method declaration. 😞

**Pull request merged**